src/engagement_db_to_analysis/data_wrangling_functions.py

```diff
@@ -0,0 +1,81 @@
+from core_data_modules.util import TimeUtils
+
+from core_data_modules.data_models.code_scheme import CodeTypes
+from core_data_modules.cleaners.cleaning_utils import CleaningUtils
+from core_data_modules.traced_data import Metadata
+from core_data_modules.logging import Logger
+
+
+log = Logger(__name__)
+
+def _impute_age_category(user, messages_traced_data, analysis_dataset_configs):
+    """
+    Imputes age category for age dataset messages.
+
+    :param user: Identifier of user running the pipeline.
+    :type user: str
+    :param messages_traced_data: Messages TracedData objects to impute age_category.
+    :type messages_traced_data: list of TracedData
+    :param analysis_dataset_config: Analysis dataset configuration in pipeline configuration module.
+    :type analysis_dataset_config: pipeline_config.analysis_configs.dataset_configurations
+    """
+
+    log.info("Imputing age categories for age dataset messages...")
+    # Get the configurations for age and age_category analysis datasets
+    updated_messages_traced_data = []
+    age_cc = None
+    age_category_cc = None
+    age_engagement_db_datasets = None
+    for analysis_dataset_config in analysis_dataset_configs:
+        for coding_config in analysis_dataset_config.coding_configs:
+            if coding_config.analysis_dataset == "age":
+                age_cc = coding_config
+                age_engagement_db_datasets = analysis_dataset_config.engagement_db_datasets
+            elif coding_config.analysis_dataset == "age_category":
+                age_category_cc = coding_config
+
+    #Check and impute age_category in age messages only
+    for message in messages_traced_data:
+        if message["dataset"] not in age_engagement_db_datasets:
+            updated_messages_traced_data.append(message)
+            continue
```

Comment:
You can check for `message["dataset"] in age_engagement_db_datasets` and then update messages traced data at the end of the for loop.

Reply:
updated thanks.